VideoLocker/src/main/java/org/edx/mobile/util/observer/AsyncCallableUtils.java

```diff
@@ -0,0 +1,37 @@
+package org.edx.mobile.util.observer;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+
+public enum AsyncCallableUtils {
+    ;
+
+    // Using AsyncTask.THREAD_POOL_EXECUTOR to automatically support Espresso idling resources
+    @VisibleForTesting
+    @NonNull
+    public static Executor EXECUTOR = AsyncTask.THREAD_POOL_EXECUTOR;
```

Comment:
Yeah, it looks like we'll need to wait to upgrade to Retrofit 2 to be able
to use OkHttp's Dispatcher class, as it requires handling it's Call
implementation directly, which is not exposed by Retrofit 1.

On Fri, Mar 18, 2016 at 12:43 AM, Brian Guertin <notifications@github.com>
wrote:

> In
> VideoLocker/src/main/java/org/edx/mobile/util/observer/AsyncCallableUtils.java
> <https://github.com/edx/edx-app-android/pull/613#discussion_r56565885>:
>
> > +package org.edx.mobile.util.observer;
> > +
> > +import android.os.AsyncTask;
> > +import android.support.annotation.NonNull;
> > +import android.support.annotation.VisibleForTesting;
> > +
> > +import java.util.concurrent.Callable;
> > +import java.util.concurrent.Executor;
> > +
> > +public enum AsyncCallableUtils {
> > +    ;
> > +
> > +    // Using AsyncTask.THREAD_POOL_EXECUTOR to automatically support Espresso idling resources
> > +    @VisibleForTesting
> > +    @NonNull
> > +    public static Executor EXECUTOR = AsyncTask.THREAD_POOL_EXECUTOR;
>
> For updating Retrofit or for replacing this class?
>
> Here, I just filed both: https://openedx.atlassian.net/browse/MA-2185
>
> I also commented in the code that it's a temporary interim solution.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/613/files/1b79f86bec593c27df395eb26a9eb9586f474ec4#r56565885>
>